packages/material/src/util/datejs.ts

```diff
@@ -30,3 +31,38 @@ export const getData = (
   }
   return dayjsData;
 };
+
+type DateInputFormEvent = FormEvent<HTMLInputElement | HTMLTextAreaElement>;
+
+/**
+ * Improves the UX of date fields by controlling the rendered input value.
+ * When a user enters a date value that ends up being different than the
+ * the value of `data`, then on blur we sync the rendered input value.
+ *
+ * @param data The parsed date value.
+ * @param onBlur Additional handler to run after input value is sync'd.
+ * @returns Props to pass to the rendered input element.
+ */
+export const useParsedDateSynchronizer = (props: {
+  data: any;
+  onBlur: FormEventHandler<HTMLInputElement | HTMLTextAreaElement> | undefined;
+}) => {
+  const [value, setValue] = useState(props.data);
+
+  const onBlur = useMemo(
+    () => (event: DateInputFormEvent) => {
+      setValue(props.data);
+      if (props.onBlur) props.onBlur(event);
+    },
+    [props.data, props.onBlur]
+  );
```

Comment:
We can use `onBlur` directly however it's already used for `focused`. So instead of injecting ourselves into `onBlur` we can just check the value of `focused`.